packages/TextEditor/src/plugins/MaxLengthPlugin.tsx

```diff
@@ -0,0 +1,51 @@
+// eslint-disable-next-line max-len
+import { useLexicalComposerContext } from '@lexical/react/LexicalComposerContext';
+import { trimTextContentFromAnchor } from '@lexical/selection';
+import { $restoreEditorState } from '@lexical/utils';
+import {
+  $getSelection,
+  $isRangeSelection,
+  EditorState,
+  RootNode,
+} from 'lexical';
+import { useEffect } from 'react';
+
+export function MaxLengthPlugin({ maxLength }: { maxLength: number }): null {
+  const [editor] = useLexicalComposerContext();
+
+  useEffect(() => {
+    let lastRestoredEditorState: EditorState | null = null;
+
+    return editor.registerNodeTransform(RootNode, (rootNode: RootNode) => {
+      const selection = $getSelection();
+      if (!$isRangeSelection(selection) || !selection.isCollapsed()) {
```

Comment:
Is collapse is a property, not a method given to lib.dom.d.ts (typescript)
![image](https://github.com/gsoft-inc/ov-igloo-ui/assets/4323800/912a6738-eead-432e-9584-ac6d6c871de1)

Reply:
It's defined in LexicalSelection as a method: 
![image](https://github.com/gsoft-inc/ov-igloo-ui/assets/110498164/dde5f303-1eca-4d5f-b583-1308cf5b46d8)
Also this was taken from lexical's playground plugins: https://github.com/facebook/lexical/blob/main/packages/lexical-playground/src/plugins/MaxLengthPlugin/index.tsx#L23